docker/base/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Use the latest Miniconda3 image with Python 3.10
+FROM continuumio/miniconda3:23.3.1-0
+
+# Build arguments to determine installation source and runhouse version to install
+ARG INSTALL_METHOD=pypi
+ARG RUNHOUSE_VERSION
+
+# Set the working directory
+WORKDIR /app
+
+# Install system dependencies
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+    rsync \
+    vim \
+    gcc \
+    screen \
+    wget \
+    curl \
+    sudo \
+    ufw \
+    git \
+    awscli && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy top-level files to /app/
+COPY setup.py pyproject.toml README.md requirements.txt pytest.ini /app/
+
+# Copy the runhouse directory that contains the app code to /app/runhouse/
+COPY runhouse/ /app/runhouse/
+
+# Determine which version of runhouse to install
+RUN if [ "$INSTALL_METHOD" = "local" ] && [ -d "/app/runhouse" ] && [ "$(ls -A /app/runhouse)" ]; then \
+      echo "Installing runhouse from local"; \
+      # Install the runhouse package with the sky dependency
+      pip install --no-cache-dir /app"[sky]"; \
+    else \
+      echo "Installing runhouse from PyPI"; \
+      if [ -z "$RUNHOUSE_VERSION" ] || [ "$RUNHOUSE_VERSION" = "latest" ]; then \
+        echo "Installing latest version"; \
+        pip install --no-cache-dir "runhouse[sky]"; \
+      else \
+        echo "Installing runhouse==${RUNHOUSE_VERSION}"; \
+        pip install --no-cache-dir "runhouse[sky]==${RUNHOUSE_VERSION}"; \
+      fi; \
+    fi
+
+# Keep the container running, allowing SkyPilot to connect and run commands inside the container
+CMD ["bash"]
```

Comment:
Is this really necessary? Feel like it's not standard

Reply:
yea fair point, it's not so standard. but we do need to keep the container up and not exit while Sky is doing the provisioning / setup step. prob better to run supervisord as the main process

Reply:
I don't really understand - isn't the Skypilot work happening after the container already built? What happens if we try to launch a docker cluster without any line here?

Reply:
Without that line (or similar) the container would exit since no process is keeping it alive. It does seem like sky is doing some work to run commands inside the container after it has started - snippet below from the provisioning logs: 

```
Error response from daemon: Container 77772152c4654088ac87318f08620c24aba7aef619fa744ff72b99668971d907 is not running

I 10-09 19:33:56 instance_setup.py:88] _initialize_docker: Retrying in 1.3 seconds, due to Command docker exec sky_container /bin/bash -c 'bash --login -c -i '"'"'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (sudo service ssh start)'"'"' failed with return code 1.

I 10-09 19:33:56 instance_setup.py:88] Failed to run docker setup commands

D 10-09 19:33:57 docker_utils.py:154] + command -v docker || echo 'NoExist'

/usr/bin/docker

D 10-09 19:33:58 docker_utils.py:154] + docker inspect -f "{{.State.Running}}" sky_container || true

false

D 10-09 19:33:58 docker_utils.py:154] + docker start sky_container

sky_container

D 10-09 19:33:59 docker_utils.py:154] + docker exec sky_container /bin/bash -c 'bash --login -c -i '"'"'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (sudo service ssh start)'"'"'

Error response from daemon: Container 77772152c4654088ac87318f08620c24aba7aef619fa744ff72b99668971d907 is not running

```